src/jit/bitsetasshortlong.h

```diff
@@ -664,9 +664,11 @@ void BitSetOps</*BitSetType*/ BitSetShortLongRep,
                /*BitSetTraits*/ BitSetTraits>::ClearDLong(Env env, BitSetShortLongRep& bs)
 {
     assert(!IsShort(env));
-    // Recall that ClearD does *not* require "bs" to be of the current epoch.
-    // Therefore, we must allocate a new representation.
-    bs = MakeEmptyArrayBits(env);
```

Comment:
> I think that in this case if we replace all ClearD with MakeEmpty we will lose a valuable information.

Yes, this is a good point. You should instead create a separate `ClearD` method with the desired semantics.

FWIW, I can attest that liveness would also be able to use the new clear method in all cases that it currently uses `ClearD`.